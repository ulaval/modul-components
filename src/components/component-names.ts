export const ACCORDION_GROUP_NAME: string = 'm-accordion-group';
export const ACCORDION_NAME: string = 'm-accordion';
export const LIST_ITEM_NAME: string = 'm-list-item';
export const BUTTON_NAME: string = 'm-button';
export const BUTTON_GROUP_NAME: string = 'm-button-group';
export const CAROUSEL_NAME: string = 'm-carousel';
export const CAROUSEL_ITEM_NAME: string = 'm-carousel-item';
export const CHECKBOX_NAME: string = 'm-checkbox';
export const DATEFIELDS_NAME: string = 'm-datefields';
export const DATEPICKER_NAME: string = 'm-datepicker';
export const DIALOG_NAME: string = 'm-dialog';
export const DROPDOWN_NAME: string = 'm-dropdown';
export const DROPDOWN_ITEM_NAME: string = 'm-dropdown-item';
export const DROPDOWN_GROUP_NAME: string = 'm-dropdown-group';
export const DYNAMIC_TEMPLATE_NAME: string = 'm-dynamic-template';
export const FILE_SELECT_NAME: string = 'm-file-select';
export const FILE_UPLOAD_NAME: string = 'm-file-upload';
export const EDIT_WINDOW_NAME: string = 'm-edit-window';
export const FLEX_TEMPLATE_NAME: string = 'm-flex-template';
export const I18N_NAME: string = 'm-i18n';
export const ICON_NAME: string = 'm-icon';
export const ICON_BUTTON_NAME: string = 'm-icon-button';
export const INPUT_STYLE_NAME: string = 'm-input-style';
export const LIMIT_TEXT_NAME: string = 'm-limit-text';
export const LINK_NAME: string = 'm-link';
export const MESSAGE_NAME: string = 'm-message';
export const MODAL_NAME: string = 'm-modal';
export const NAVBAR_NAME: string = 'm-navbar';
export const NAVBAR_ITEM_NAME: string = 'm-navbar-item';
export const MENU_NAME: string = 'm-menu';
export const MENU_ITEM_NAME: string = 'm-menu-item';
export const PANEL_NAME: string = 'm-panel';
export const PHONE_NUMBER_NAME: string = 'm-phone-number';
export const POPPER_NAME: string = 'm-popper';
export const POPUP_NAME: string = 'm-popup';
export const PROGRESS_NAME: string = 'm-progress';
export const RADIO_NAME: string = 'm-radio';
export const RADIO_GROUP_NAME: string = 'm-radio-group';
export const RADIO_STYLE_NAME: string = 'm-radio-style';
export const SCROLL_TOP_NAME: string = 'm-scroll-top';
export const SIDEBAR_NAME: string = 'm-sidebar';
export const SLIDER_NAME: string = 'm-slider';
export const SPINNER_NAME: string = 'm-spinner';
export const STATUS_NAME: string = 'm-status';
export const STEP_NAME: string = 'm-step';
export const STEPPERS_NAME: string = 'm-steppers';
export const STEPPERS_ITEM_NAME: string = 'm-steppers-item';
export const SWITCH_NAME: string = 'm-switch';
export const TAB_PANEL_NAME: string = 'm-tab-panel';
export const TABS_NAME: string = 'm-tabs';
export const TEMPLATE_NAME: string = 'm-template';
export const TEXTAREA_NAME: string = 'm-textarea';
export const TEXTFIELD_NAME: string = 'm-textfield';
export const TIMEPICKER_NAME: string = 'm-timepicker';
export const TOOLTIP_NAME: string = 'm-tooltip';
export const UPLOAD_NAME: string = 'm-upload';
export const UPLOAD_INPUT_NAME: string = 'm-upload-input';
export const UPLOAD_DRAGDROP_NAME: string = 'm-upload-dragdrop';
export const UPLOAD_FILESLIST_NAME: string = 'm-upload-fileslist';
export const VALIDATION_MESSAGE_NAME: string = 'm-validation-message';
