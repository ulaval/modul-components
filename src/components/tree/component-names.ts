export const TREE_NODE_NAME: string = 'm-tree-node';
export const TREE_ICON_NAME: string = 'm-tree-icon';
