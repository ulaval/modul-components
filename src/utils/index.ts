import Vue, { PluginObject } from 'vue';

import MediaQueriesPlugin from './media-queries/media-queries';
import ModulPlugin from './modul/modul';
import MessagesPlugin from './i18n/i18n';
import HttpPlugin, { HttpPluginOptions } from './http/http';
import SpritesPlugin from './svg/sprites';
import * as TouchPlugin from 'vue-touch';
import ConfirmPlugin from './modal/confirm';
import AlertPlugin from './modal/alert';
import FilePlugin from '../utils/file/file';
import { PortalPluginInstall } from 'portal-vue';
import LoggerPlugin, { ConsoleOptions } from './logger/logger';

export interface UtilsPluginOptions {
    httpPluginOptions?: HttpPluginOptions;
    consoleOptions?: ConsoleOptions;
}

const UtilsPlugin: PluginObject<any> = {
    install(v, options): void {
        if (!v.prototype.$log) {
            Vue.use(LoggerPlugin, options ? options.consoleOptions : undefined);
        } else if (options) {
            v.prototype.$log.setConsoleOptions(options.consoleOptions);
        }

        Vue.use(MediaQueriesPlugin);
        Vue.use(ModulPlugin);
        Vue.use(MessagesPlugin);
        Vue.use(HttpPlugin, options ? options.httpPluginOptions : undefined);
        Vue.use({ install: PortalPluginInstall });
        Vue.use(SpritesPlugin);
        Vue.use(TouchPlugin, { name: 'v-touch' });
        Vue.use(ConfirmPlugin);
        Vue.use(AlertPlugin);
        Vue.use(FilePlugin);
    }
};

export default UtilsPlugin;
