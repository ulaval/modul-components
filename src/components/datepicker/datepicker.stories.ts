import { withA11y } from '@storybook/addon-a11y';
import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import Vue from 'vue';
import { componentsHierarchyRootSeparator } from '../../../conf/storybook/utils';
import { DATEPICKER_NAME } from '../component-names';
import DatepickerPlugin from './datepicker';

Vue.use(DatepickerPlugin);

declare module '@storybook/addon-knobs' {
    export function withKnobs(): any;
}

storiesOf(`${componentsHierarchyRootSeparator}${DATEPICKER_NAME}`, module)
    .addDecorator(withA11y)
    .addDecorator(withKnobs)
    .add('default', () => ({
        template: `<m-datepicker></m-datepicker>`
    }))
    .add('events', () => ({
        data: () => ({
            model1: '2011-01-01'
        }),
        methods: {
            onInputChange(value: string): string {
                // tslint:disable-next-line: no-console
                console.log('MDatePicker.onInputChange=' + value);
                return value;
            },
            onFocus(value: Event): void {
                // tslint:disable-next-line: no-console
                console.log('MDatePicker.onFocus!');

            },
            onBlur(event: Event): void {
                // tslint:disable-next-line: no-console
                console.log('MDatePicker.onBlur!');

            }
        },
        template: `<div><m-datepicker :value="model1" @change="model1 = onInputChange($event)" @focus="onFocus" @blur="onBlur"></m-datepicker> <br/><br/>model value = {{model1}}</div>`
    }))
    .add('label', () => ({
        template: `<m-datepicker label="Date label"></m-datepicker>`
    }))

    .add('placeholder', () => ({
        template: `<m-datepicker placeholder="Lorem Ipsum"></m-datepicker>`
    }))

    .add('waiting', () => ({
        template: `<m-datepicker :waiting="true"></m-datepicker>`
    }))

    .add('min="2008-01-01" && max="2014-12-31"', () => ({
        data: () => ({
            model1: '2011-01-01'
        }),
        template: `<div><m-datepicker min="2008-01-01" max="2014-12-31" v-model="model1"></m-datepicker>model value = {{model1}}</div>`
    }))
    .add('date format invalid', () => ({
        data: () => ({
            model1: '2000-19-12'
        }),
        template: `<div><m-datepicker min="2008-01-01" max="2014-12-31" v-model="model1"></m-datepicker>model value = {{model1}}</div>`
    }))
    .add('date off limit min', () => ({
        data: () => ({
            model1: '2000-01-01'
        }),
        template: `<div><m-datepicker min="2008-01-01" max="2014-12-31" v-model="model1"></m-datepicker>model value = {{model1}}</div>`
    }))
    .add('date off limit  max', () => ({
        data: () => ({
            model1: '2015-01-01'
        }),
        template: `<div><m-datepicker min="2008-01-01" max="2014-12-31" v-model="model1"></m-datepicker>model value = {{model1}}</div>`
    }))

    .add('disabled', () => ({
        template: `<m-datepicker :disabled="true"></m-datepicker>`
    }))
    .add('readonly', () => ({
        template: `<m-datepicker :readonly="true"></m-datepicker>`
    }))
    .add('valid', () => ({
        template: `<m-datepicker :valid="true"></m-datepicker>`
    }))
    .add('error-message', () => ({
        template: `<m-datepicker :error="true" error-message="this is an error"></m-datepicker>`
    }))
    .add('helper-message', () => ({
        template: `<m-datepicker helper-message="AAAA-MM-JJ"></m-datepicker>`
    }))
    .add('hide-internal-error-message', () => ({
        data: () => ({
            model1: '9999-99-99'
        }),
        template: `<m-datepicker v-model="model1" :hide-internal-error-message="true"></m-datepicker>`
    }));

