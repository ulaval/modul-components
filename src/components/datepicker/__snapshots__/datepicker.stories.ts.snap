// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components|m-datepicker date format invalid 1`] = `
<div>
  <div
    class="m-datepicker m--has-error"
    style="width: 100%; max-width: 136px;"
  >
    <div
      class="m-input-style m--has-error m--has-value m--is-tag-default"
    >
      <div
        class="m-input-style__content"
        style="margin-top: 10px;"
      >
        <!---->
         
        <div
          class="m-input-style__body"
        >
          <div
            class="m-input-style__prefix"
          />
           
          <div
            class="m-input-style__left-content"
          >
            <input
              aria-haspopup="true"
              autocomplete="off"
              id="mDatepicker-uuid"
              placeholder="AAAA-MM-JJ"
            />
              
            <!---->
             
            <!---->
          </div>
           
          <div
            class="m-input-style__right-content"
          >
            <button
              aria-controls="mDatepicker-uuid"
              class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
              style="width: 22px; height: 22px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="m-icon m-icon-button__icon"
                height="20px"
                width="20px"
              >
                <!---->
                 
                <use
                  aria-hidden="true"
                  class=""
                />
              </svg>
               
              <!---->
            </button>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="m-validation-message m-datepicker__validation-message"
    >
      <p
        class="m-validation-message__error"
      >
        <svg
          class="m-icon m-validation-message__icon"
          height="1em"
          width="1em"
        >
          <title>
            Erreur
          </title>
           
          <use
            aria-hidden="true"
            class=""
          />
        </svg>
         
        <span
          class="m-validation-message__text"
        >
          Format invalide
        </span>
      </p>
       
      <!---->
    </div>
     
    <div
      class="m-popup"
    >
      <div
        class="m-popper"
      >
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
  model value = 2000-19-12
</div>
`;

exports[`Storyshots components|m-datepicker date off limit  max 1`] = `
<div>
  <div
    class="m-datepicker m--has-error"
    style="width: 100%; max-width: 136px;"
  >
    <div
      class="m-input-style m--has-error m--has-value m--is-tag-default"
    >
      <div
        class="m-input-style__content"
        style="margin-top: 10px;"
      >
        <!---->
         
        <div
          class="m-input-style__body"
        >
          <div
            class="m-input-style__prefix"
          />
           
          <div
            class="m-input-style__left-content"
          >
            <input
              aria-haspopup="true"
              autocomplete="off"
              id="mDatepicker-uuid"
              placeholder="AAAA-MM-JJ"
            />
              
            <!---->
             
            <!---->
          </div>
           
          <div
            class="m-input-style__right-content"
          >
            <button
              aria-controls="mDatepicker-uuid"
              class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
              style="width: 22px; height: 22px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="m-icon m-icon-button__icon"
                height="20px"
                width="20px"
              >
                <!---->
                 
                <use
                  aria-hidden="true"
                  class=""
                />
              </svg>
               
              <!---->
            </button>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="m-validation-message m-datepicker__validation-message"
    >
      <p
        class="m-validation-message__error"
      >
        <svg
          class="m-icon m-validation-message__icon"
          height="1em"
          width="1em"
        >
          <title>
            Erreur
          </title>
           
          <use
            aria-hidden="true"
            class=""
          />
        </svg>
         
        <span
          class="m-validation-message__text"
        >
          Date hors limites
        </span>
      </p>
       
      <!---->
    </div>
     
    <div
      class="m-popup"
    >
      <div
        class="m-popper"
      >
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
  model value = 2015-01-01
</div>
`;

exports[`Storyshots components|m-datepicker date off limit min 1`] = `
<div>
  <div
    class="m-datepicker m--has-error"
    style="width: 100%; max-width: 136px;"
  >
    <div
      class="m-input-style m--has-error m--has-value m--is-tag-default"
    >
      <div
        class="m-input-style__content"
        style="margin-top: 10px;"
      >
        <!---->
         
        <div
          class="m-input-style__body"
        >
          <div
            class="m-input-style__prefix"
          />
           
          <div
            class="m-input-style__left-content"
          >
            <input
              aria-haspopup="true"
              autocomplete="off"
              id="mDatepicker-uuid"
              placeholder="AAAA-MM-JJ"
            />
              
            <!---->
             
            <!---->
          </div>
           
          <div
            class="m-input-style__right-content"
          >
            <button
              aria-controls="mDatepicker-uuid"
              class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
              style="width: 22px; height: 22px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="m-icon m-icon-button__icon"
                height="20px"
                width="20px"
              >
                <!---->
                 
                <use
                  aria-hidden="true"
                  class=""
                />
              </svg>
               
              <!---->
            </button>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="m-validation-message m-datepicker__validation-message"
    >
      <p
        class="m-validation-message__error"
      >
        <svg
          class="m-icon m-validation-message__icon"
          height="1em"
          width="1em"
        >
          <title>
            Erreur
          </title>
           
          <use
            aria-hidden="true"
            class=""
          />
        </svg>
         
        <span
          class="m-validation-message__text"
        >
          Date hors limites
        </span>
      </p>
       
      <!---->
    </div>
     
    <div
      class="m-popup"
    >
      <div
        class="m-popper"
      >
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
  model value = 2000-01-01
</div>
`;

exports[`Storyshots components|m-datepicker default 1`] = `
<div
  class="m-datepicker"
  style="width: 100%; max-width: 136px;"
>
  <div
    class="m-input-style m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 10px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
          <input
            aria-haspopup="true"
            autocomplete="off"
            id="mDatepicker-uuid"
            placeholder="AAAA-MM-JJ"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        >
          <button
            aria-controls="mDatepicker-uuid"
            class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
            style="width: 22px; height: 22px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="20px"
              width="20px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-validation-message m-datepicker__validation-message"
  >
    <!---->
     
    <!---->
  </div>
   
  <div
    class="m-popup"
  >
    <div
      class="m-popper"
    >
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots components|m-datepicker disabled 1`] = `
<div
  class="m-datepicker m--is-disabled"
  style="width: 100%; max-width: 136px;"
>
  <div
    class="m-input-style m--is-disabled m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 10px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
          <input
            aria-haspopup="true"
            autocomplete="off"
            disabled="disabled"
            id="mDatepicker-uuid"
            placeholder="AAAA-MM-JJ"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        >
          <button
            aria-controls="mDatepicker-uuid"
            class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple m--is-disabled"
            disabled="disabled"
            style="width: 22px; height: 22px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="20px"
              width="20px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-validation-message m-datepicker__validation-message"
  >
    <!---->
     
    <!---->
  </div>
   
  <div
    class="m-popup"
  >
    <div
      class="m-popper"
    >
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots components|m-datepicker error-message 1`] = `
<div
  class="m-datepicker m--has-error m--has-validation-message"
  style="width: 100%; max-width: 136px;"
>
  <div
    class="m-input-style m--has-error m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 10px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
          <input
            aria-haspopup="true"
            autocomplete="off"
            id="mDatepicker-uuid"
            placeholder="AAAA-MM-JJ"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        >
          <button
            aria-controls="mDatepicker-uuid"
            class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
            style="width: 22px; height: 22px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="20px"
              width="20px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-validation-message m-datepicker__validation-message"
  >
    <p
      class="m-validation-message__error"
    >
      <svg
        class="m-icon m-validation-message__icon"
        height="1em"
        width="1em"
      >
        <title>
          Erreur
        </title>
         
        <use
          aria-hidden="true"
          class=""
        />
      </svg>
       
      <span
        class="m-validation-message__text"
      >
        this is an error
      </span>
    </p>
     
    <!---->
  </div>
   
  <div
    class="m-popup"
  >
    <div
      class="m-popper"
    >
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots components|m-datepicker events 1`] = `
<div>
  <div
    class="m-datepicker"
    style="width: 100%; max-width: 136px;"
  >
    <div
      class="m-input-style m--has-value m--is-tag-default"
    >
      <div
        class="m-input-style__content"
        style="margin-top: 10px;"
      >
        <!---->
         
        <div
          class="m-input-style__body"
        >
          <div
            class="m-input-style__prefix"
          />
           
          <div
            class="m-input-style__left-content"
          >
            <input
              aria-haspopup="true"
              autocomplete="off"
              id="mDatepicker-uuid"
              placeholder="AAAA-MM-JJ"
            />
              
            <!---->
             
            <!---->
          </div>
           
          <div
            class="m-input-style__right-content"
          >
            <button
              aria-controls="mDatepicker-uuid"
              class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
              style="width: 22px; height: 22px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="m-icon m-icon-button__icon"
                height="20px"
                width="20px"
              >
                <!---->
                 
                <use
                  aria-hidden="true"
                  class=""
                />
              </svg>
               
              <!---->
            </button>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="m-validation-message m-datepicker__validation-message"
    >
      <!---->
       
      <!---->
    </div>
     
    <div
      class="m-popup"
    >
      <div
        class="m-popper"
      >
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
   
  <br />
  <br />
  model value = 2011-01-01
</div>
`;

exports[`Storyshots components|m-datepicker helper-message 1`] = `
<div
  class="m-datepicker m--has-validation-message"
  style="width: 100%; max-width: 136px;"
>
  <div
    class="m-input-style m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 10px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
          <input
            aria-haspopup="true"
            autocomplete="off"
            id="mDatepicker-uuid"
            placeholder="AAAA-MM-JJ"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        >
          <button
            aria-controls="mDatepicker-uuid"
            class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
            style="width: 22px; height: 22px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="20px"
              width="20px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-validation-message m-datepicker__validation-message"
  >
    <!---->
     
    <p
      class="m-validation-message__helper"
    >
      AAAA-MM-JJ
    </p>
  </div>
   
  <div
    class="m-popup"
  >
    <div
      class="m-popper"
    >
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots components|m-datepicker hide-internal-error-message 1`] = `
<div
  class="m-datepicker m--has-error"
  style="width: 100%; max-width: 136px;"
>
  <div
    class="m-input-style m--has-error m--has-value m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 10px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
          <input
            aria-haspopup="true"
            autocomplete="off"
            id="mDatepicker-uuid"
            placeholder="AAAA-MM-JJ"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        >
          <button
            aria-controls="mDatepicker-uuid"
            class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
            style="width: 22px; height: 22px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="20px"
              width="20px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-validation-message m-datepicker__validation-message"
  >
    <!---->
     
    <!---->
  </div>
   
  <div
    class="m-popup"
  >
    <div
      class="m-popper"
    >
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots components|m-datepicker label 1`] = `
<div
  class="m-datepicker"
  style="width: 100%; max-width: 136px;"
>
  <div
    class="m-input-style m--has-label m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 10px;"
    >
      <label
        class="m-input-style__label"
        for="mDatepicker-uuid"
      >
        <span
          class="m-input-style__text"
        >
          <span>
            Date label
          </span>
           
          <!---->
        </span>
      </label>
       
      <div
        class="m-input-style__body"
      >
        <!---->
         
        <div
          class="m-input-style__left-content"
        >
          <input
            aria-haspopup="true"
            autocomplete="off"
            id="mDatepicker-uuid"
            placeholder="AAAA-MM-JJ"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        >
          <button
            aria-controls="mDatepicker-uuid"
            class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
            style="width: 22px; height: 22px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="20px"
              width="20px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-validation-message m-datepicker__validation-message"
  >
    <!---->
     
    <!---->
  </div>
   
  <div
    class="m-popup"
  >
    <div
      class="m-popper"
    >
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots components|m-datepicker min="2008-01-01" && max="2014-12-31" 1`] = `
<div>
  <div
    class="m-datepicker"
    style="width: 100%; max-width: 136px;"
  >
    <div
      class="m-input-style m--has-value m--is-tag-default"
    >
      <div
        class="m-input-style__content"
        style="margin-top: 10px;"
      >
        <!---->
         
        <div
          class="m-input-style__body"
        >
          <div
            class="m-input-style__prefix"
          />
           
          <div
            class="m-input-style__left-content"
          >
            <input
              aria-haspopup="true"
              autocomplete="off"
              id="mDatepicker-uuid"
              placeholder="AAAA-MM-JJ"
            />
              
            <!---->
             
            <!---->
          </div>
           
          <div
            class="m-input-style__right-content"
          >
            <button
              aria-controls="mDatepicker-uuid"
              class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
              style="width: 22px; height: 22px;"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="m-icon m-icon-button__icon"
                height="20px"
                width="20px"
              >
                <!---->
                 
                <use
                  aria-hidden="true"
                  class=""
                />
              </svg>
               
              <!---->
            </button>
          </div>
           
          <!---->
        </div>
      </div>
    </div>
     
    <div
      class="m-validation-message m-datepicker__validation-message"
    >
      <!---->
       
      <!---->
    </div>
     
    <div
      class="m-popup"
    >
      <div
        class="m-popper"
      >
         
        <!---->
      </div>
       
      <!---->
    </div>
  </div>
  model value = 2011-01-01
</div>
`;

exports[`Storyshots components|m-datepicker placeholder 1`] = `
<div
  class="m-datepicker"
  style="width: 100%; max-width: 136px;"
>
  <div
    class="m-input-style m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 10px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
          <input
            aria-haspopup="true"
            autocomplete="off"
            id="mDatepicker-uuid"
            placeholder="Lorem Ipsum"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        >
          <button
            aria-controls="mDatepicker-uuid"
            class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
            style="width: 22px; height: 22px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="20px"
              width="20px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-validation-message m-datepicker__validation-message"
  >
    <!---->
     
    <!---->
  </div>
   
  <div
    class="m-popup"
  >
    <div
      class="m-popper"
    >
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots components|m-datepicker readonly 1`] = `
<div
  class="m-datepicker m--is-readonly"
  style="width: 100%; max-width: 136px;"
>
  <div
    class="m-input-style m--is-readonly m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 10px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
          <input
            aria-haspopup="true"
            autocomplete="off"
            id="mDatepicker-uuid"
            placeholder="AAAA-MM-JJ"
            readonly="readonly"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        >
          <button
            aria-controls="mDatepicker-uuid"
            class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple m--is-disabled"
            disabled="disabled"
            style="width: 22px; height: 22px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="20px"
              width="20px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-validation-message m-datepicker__validation-message"
  >
    <!---->
     
    <!---->
  </div>
   
  <div
    class="m-popup"
  >
    <div
      class="m-popper"
    >
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots components|m-datepicker valid 1`] = `
<div
  class="m-datepicker m--is-valid"
  style="width: 100%; max-width: 136px;"
>
  <div
    class="m-input-style m--is-valid m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 10px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
          <input
            aria-haspopup="true"
            autocomplete="off"
            id="mDatepicker-uuid"
            placeholder="AAAA-MM-JJ"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        >
          <button
            aria-controls="mDatepicker-uuid"
            class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple"
            style="width: 22px; height: 22px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="20px"
              width="20px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
         
        <!---->
      </div>
    </div>
  </div>
   
  <div
    class="m-validation-message m-datepicker__validation-message"
  >
    <!---->
     
    <!---->
  </div>
   
  <div
    class="m-popup"
  >
    <div
      class="m-popper"
    >
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`Storyshots components|m-datepicker waiting 1`] = `
<div
  class="m-datepicker m--is-waiting"
  style="width: 100%; max-width: 136px;"
>
  <div
    class="m-input-style m--is-waiting m--is-tag-default"
  >
    <div
      class="m-input-style__content"
      style="margin-top: 10px;"
    >
      <!---->
       
      <div
        class="m-input-style__body"
      >
        <div
          class="m-input-style__prefix"
        />
         
        <div
          class="m-input-style__left-content"
        >
          <input
            aria-haspopup="true"
            autocomplete="off"
            disabled="disabled"
            id="mDatepicker-uuid"
            placeholder="AAAA-MM-JJ"
          />
            
          <!---->
           
          <!---->
        </div>
         
        <div
          class="m-input-style__right-content"
        >
          <button
            aria-controls="mDatepicker-uuid"
            class="m-icon-button m-datepicker__icon m--is-skin-light m--has-ripple m--is-disabled"
            disabled="disabled"
            style="width: 22px; height: 22px;"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-icon-button__icon"
              height="20px"
              width="20px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <!---->
          </button>
        </div>
         
        <div
          class="m-spinner m-input-style__spinner"
        >
          <!---->
           
          <div />
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="m-validation-message m-datepicker__validation-message"
  >
    <!---->
     
    <!---->
  </div>
   
  <div
    class="m-popup"
  >
    <div
      class="m-popper"
    >
       
      <!---->
    </div>
     
    <!---->
  </div>
</div>
`;
